transport/src/test/java/io/netty/channel/DefaultChannelPipelineTest.java

```diff
@@ -1073,6 +1073,38 @@ public void testNotPinExecutor() {
         group.shutdownGracefully(0, 0, TimeUnit.SECONDS);
     }
 
+    @Test
+    public void testHandlerAfterClose() throws Exception {
+        EmbeddedChannel channel = new EmbeddedChannel();
+        channel.close().syncUninterruptibly();
+
+        assertFalse(channel.isActive());
+
+        final AtomicBoolean addedHandler = new AtomicBoolean();
+        final AtomicBoolean removedHandler = new AtomicBoolean();
+
+        ChannelPipeline pipeline = channel.pipeline();
+        try {
+            pipeline.addLast(new ChannelHandlerAdapter() {
+                @Override
+                public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
+                    addedHandler.set(true);
+                }
+
+                @Override
+                public void handlerRemoved(ChannelHandlerContext ctx) throws Exception {
+                    removedHandler.set(true);
+                }
+            });
+            fail();
+        } catch (ChannelPipelineException e) {
+            // expected
```

Comment:
if you happen to revisit this, it would be nice to ensure that we enter this catch block